cli/azd/pkg/project/container_helper.go

```diff
@@ -440,20 +496,48 @@ func (ch *ContainerHelper) runRemoteBuild(
 		return "", fmt.Errorf("remote build is only supported when the target registry is an Azure Container Registry")
 	}
 
-	registryResourceName := strings.TrimSuffix(registryName, acrRegistryDomain)
+	localImageTag, err := ch.LocalImageTag(ctx, serviceConfig)
+	if err != nil {
+		return "", err
+	}
 
-	source, err := ch.remoteBuildManager.UploadBuildSource(
-		ctx, target.SubscriptionId(), target.ResourceGroupName(), registryResourceName, contextPath)
+	imageName, err := ch.RemoteImageTag(ctx, serviceConfig, localImageTag, options)
 	if err != nil {
 		return "", err
 	}
 
-	localImageTag, err := ch.LocalImageTag(ctx, serviceConfig)
+	// `azd deploy` skips building if the image already exists in the registry
+	// `azd publish` always overwrites if the image already exists in the registry
+	if options == nil || !options.Overwrite {
```

Comment:
Can you imagine a way forward to separate the logic out such that this logic wouldn't be "dual-purposed"?

Reply:
Do you mind clarifying? Currently the `Overwrite` option is set in `deploy.go` and `publish.go`

Reply:
With the current implementation, `Publish()` says pushes and image to a remote server, *only if the remote image doesn't already exist*. Technically, the API Is `PublishIfNotAlreadyExists()`.

Another thing that adds to the API behavioral complexity is that the logic is actually embedded deep in the bowels of `runLocalBuild` and `runRemoteBuild`. Sometimes, there are good reasons to do this -- performance is usually a good one. But here, I suspect we're simply following the path of least resistance. 

I am still getting a deeper look into a better overall recommendation for a refactor.